src/Pages/Signup/Signup.js

```diff
@@ -1,12 +1,371 @@
 import React, { Component } from "react";
+import "./Signup.scss";
 
 class Signup extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      isModalActive: false,
+      showIdDetail: false,
+      showPwDetail: false,
+      showCheckPwDetail: false,
+      validateId: false,
+      validatePw: false,
+      validatePwLength: false,
+      isPwSame: false,
+      account: "",
+      password: "",
+      name: "",
+      email: "",
+    };
   }
+
+  //this.setState({ info.account: })
+
+  handleModal = () => {
+    this.setState({ isModalActive: !this.state.isModalActive });
+  };
+
+  openIdDetail = () => {
+    this.setState({ showIdDetail: true });
+  };
+
+  openPwDetail = () => {
+    this.setState({ showPwDetail: true });
+  };
+
+  openCheckPwDetail = () => {
+    this.setState({ showCheckPwDetail: true });
+  };
+
+  validationId = e => {
+    const { value } = e.target;
+    this.setState({ account: value });
+    const txt = /^[a-z]{6,}$/;
+    const numtxt6 = /^(?=.*[0-9])(?=.*[a-zA-Z]).{6,}$/;
+    const both = txt.test(value) || numtxt6.test(value);
```

Comment:
아래와 같이 작성하는게 더 가독성에 좋을 것 같습니다. 
```suggestion
    const { value } = e.target;
    const txt = /^[a-z]{6,}$/;
    const numtxt6 = /^(?=.*[0-9])(?=.*[a-zA-Z]).{6,}$/;
    const both = txt.test(value) || numtxt6.test(value);
    this.setState({ account: value });
```

Reply:
네 그런것 같아요 !! 감사합니다 💕